ryhope/src/storage/updatetree.rs

```diff
@@ -168,20 +139,146 @@ impl<K: Debug + Clone + Hash + Eq + Sync + Send> UpdateTree<K> {
     pub fn extend_with_path(&mut self, mut path: Vec<K>) {
         path.reverse();
         if let Some(k) = path.pop() {
-            assert_eq!(k, self.nodes[0].k);
+            assert_eq!(k, self.node(0).k);
             self.rec_from_path(0, path);
         }
     }
 
     /// Create a workplan from this tree, consuming it.
     pub fn into_workplan(self) -> UpdatePlan<K> {
-        UpdatePlan::new(self)
+        UpdatePlan::new(self, 1)
+    }
+
+    /// Create a workplan of subtrees up to `batch_size` elements from this
+    /// tree, consuming it.
+    pub fn into_batched_workplan(self, subtree_size: usize) -> UpdatePlan<K> {
```

Comment:
Does it make sense to give an upper bound as well ? Like if it have the following tree
* leaves: A B C D
* parent E F
* root G
and I give subtree size 4, then what is the result ? A substree containing A B E + .. ? F C D  and thus G ? i.e. the whole subtree ? 
My point is there a unique size of number of nodes in the subtree might lead to undesired behavior, wdyt ?

Reply:
It would be `(E (A B))` (going up to `G` would be size 7 > 4), then `(G (F (C D)))`.

Is that the desired behavior?

Reply:
Ah ok good so the subtree size is already an upper bound. good then 👍  
(i should have read the comments on top of the function first... sorry >< )